ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -8452,30 +8452,34 @@ void HqlCppTranslator::optimizeOrderValues(HqlExprArray & leftValues, HqlExprArr
     {
         IHqlExpression * curFirstLeft = &leftValues.item(i);
         IHqlExpression * curFirstRight = &rightValues.item(i);
-        IHqlExpression * leftSel = querySimpleOrderSelector(curFirstLeft);
-        IHqlExpression * rightSel = querySimpleOrderSelector(curFirstRight);
+        bool leftIsNew;
+        bool rightIsNew;
+        IHqlExpression * leftSel = querySimpleOrderSelector(curFirstLeft, leftIsNew);
+        IHqlExpression * rightSel = querySimpleOrderSelector(curFirstRight, rightIsNew);
         if (!leftSel || !rightSel)
             continue;
 
         unsigned compareSize = getMemcmpSize(curFirstLeft, curFirstRight, isEqualityCompare);
         if (!compareSize)
             continue;
 
+        bool nextLeftIsNew;
+        bool nextRightIsNew;
         IHqlExpression * nextLeft = &leftValues.item(i+1);
         IHqlExpression * nextRight = &rightValues.item(i+1);
-        if (querySimpleOrderSelector(nextLeft) != leftSel || querySimpleOrderSelector(nextRight) != rightSel ||
+        if (querySimpleOrderSelector(nextLeft, nextLeftIsNew) != leftSel || querySimpleOrderSelector(nextRight, nextRightIsNew) != rightSel ||
```

Comment:
If the first returns true, the second won't be called and nextRightIsNew won't be set?

Reply:
That's ok - it goes back to the start of the loop if that is true, and never uses those variables.